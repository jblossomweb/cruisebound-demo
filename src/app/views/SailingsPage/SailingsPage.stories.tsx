import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import SailingsPage from './SailingsPage'

export default {
  title: 'views/SailingsPage',
  component: SailingsPage,
} as Meta<typeof SailingsPage>

const Template: StoryFn<typeof SailingsPage> = () => <SailingsPage />

export const Sample = Template.bind({})
