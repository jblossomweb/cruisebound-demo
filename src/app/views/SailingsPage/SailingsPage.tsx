import React from 'react'
import type { NextPage } from 'next'
import Head from 'next/head'

import type Sailing from 'app/types/Sailing'
import useFilters from 'app/hooks/useFilters'

import FoldingColumn from 'app/templates/FoldingColumn'
import SailingFilters from 'app/components/molecules/SailingFilters'
import SailingList from 'app/components/organisms/SailingList'

import styles from './SailingsPage.module.css'

export interface SailingsPageProps {
  sailings: Sailing[]
  error?: Error | null
}

const SailingsPage: NextPage<SailingsPageProps> = ({
  sailings = [],
  error = undefined,
}: SailingsPageProps) => {
  const withIds = sailings
    // assign sequential ids to full list for the key prop
    .map((sailing, i) => ({
      id: i,
      ...sailing,
    }))

  const {
    portOptions,
    cruiselineOptions,
    filterPort,
    filterCruiseline,
    setFilterCruiseline,
    setFilterPort,
    resetFilters,
    filtering,
    isFiltered,
    filteredResult,
  } = useFilters(withIds)

  return (
    <div className={styles.container}>
      <Head>
        <title>Cruisebound Demo - Sailings</title>
        <meta
          name="description"
          content="Sail away, sail away sail away sail away"
        />
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <FoldingColumn
        leftContent={
          <SailingFilters
            portOptions={portOptions}
            cruiselineOptions={cruiselineOptions}
            selectedPort={filterPort}
            selectedCruiseline={filterCruiseline}
            onPortChange={setFilterPort}
            onCruiselineChange={setFilterCruiseline}
          />
        }
        rightContent={
          <SailingList
            filtering={filtering}
            sailings={filteredResult}
            error={error || undefined}
            showReset={isFiltered}
            onReset={resetFilters}
          />
        }
      />
    </div>
  )
}

export default SailingsPage
