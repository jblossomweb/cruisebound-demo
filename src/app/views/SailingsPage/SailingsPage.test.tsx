/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import Index from './index'
import * as stories from './SailingsPage.stories'

describe('app/views/SailingsPage', () => {
  describe('index', () => {
    afterEach(cleanup)

    it('should wireup without crashing', () => {
      render(<Index />)
    })
  })

  describe('component', () => {
    describe('stories', () => {
      storySnaps(stories)
    })
  })
})
