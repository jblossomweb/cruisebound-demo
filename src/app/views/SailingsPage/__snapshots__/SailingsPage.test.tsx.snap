// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/views/SailingsPage component stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="page"
    >
      <div
        class="left expanded"
        data-testid="left-col"
      >
        <div
          class="content"
        >
          <div
            class="stack"
          >
            <div
              class="top"
            >
              <div
                class="toggle"
              >
                <button
                  class="btn btn-slate btn-medium"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 9 9"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M1 4.5L8.5 4.5"
                      stroke="#fff"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                    <path
                      d="M4.75 0.750001L8.5 4.5L4.75 8.25"
                      stroke="#fff"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="main"
              >
                <div
                  class="filters"
                >
                  <div
                    class="filter"
                  >
                    <label
                      class="label"
                    >
                      Departure port
                    </label>
                    <div
                      class="select wide naked"
                    >
                      <select
                        class="inner"
                        data-testid="select"
                      >
                        <option
                          value="any"
                        >
                          Any port
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="filter"
                  >
                    <label
                      class="label"
                    >
                      Cruiseline
                    </label>
                    <div
                      class="select wide naked"
                    >
                      <select
                        class="inner"
                        data-testid="select"
                      >
                        <option
                          value="any"
                        >
                          Any ship
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="footer"
            >
              <figure
                class=""
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27176%27%20height=%2786%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Cruisebound"
                    aria-label="Cruisebound"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    title="Cruisebound"
                  />
                  <noscript />
                </span>
              </figure>
            </div>
          </div>
        </div>
      </div>
      <div
        class="right"
      >
        <div
          class="content"
        >
          <main
            class="main"
          >
            <div
              class="sort"
            >
              <div
                class="sorters"
              >
                <label
                  class="label"
                >
                  Sort by
                </label>
                <div
                  class="select shadow"
                >
                  <select
                    class="inner"
                    data-testid="select"
                  >
                    <option
                      value="rating"
                    >
                      Rating (highest first)
                    </option>
                    <option
                      value="price"
                    >
                      Price (lowest first)
                    </option>
                    <option
                      value="duration"
                    >
                      Duration (longest first)
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div
              role="status"
            >
              <svg
                aria-hidden="true"
                class="small blue spinner"
                fill="none"
                viewBox="0 0 100 101"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z"
                  fill="currentColor"
                />
                <path
                  d="M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z"
                  fill="currentFill"
                />
              </svg>
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </main>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
